File: java/iterator/ej1/ConcreteAggregate.java
package patrones.iterator.ej1;

import java.util.ArrayList;

class ConcreteAggregate extends Aggregate
{
    private ArrayList<Object> items = new ArrayList<Object>();

    public Iterator createIterator()
    {
        return new ConcreteIterator(this);
    }

    public int count()
    {
        return items.size(); 
    }

    public Object getItem(int index)
    {
        return items.get(index);
    }
    public void setItem(int index, Object value){
    	items.add(index,value);
    	
    	
    }
}

File: c#/iterator/Ejm1/ConcreteAggregate.cs
using System.Collections;
namespace Iterator.Ejm1
{
    /// <summary>
    /// Implementa la interfaz de creación de Iterator para devolver una instancia ConcreteIterator apropiada
    /// </summary>
    class ConcreteAggregate : Aggregate
    {
        private ArrayList items = new ArrayList();

        public override Iterator CreateIterator()
        {
            return new ConcreteIterator(this);
        }

        // Property 
        public int Count
        {
            get { return items.Count; }
        }

        // Indexer 
        public object this[int index]
        {
            get { return items[index]; }
            set { items.Insert(index, value); }
        }
    }
}
